plugin-server/src/main/ingestion-queues/session-recording/blob-ingester/session-manager.ts

```diff
@@ -157,15 +158,20 @@ export class SessionManager {
             return
         }
 
-        const bufferAge = referenceNow - this.buffer.oldestKafkaTimestamp
+        const bufferAgeInMemory = now() - this.buffer.createdAt
+        const bufferAgeFromReference = referenceNow - this.buffer.oldestKafkaTimestamp
+
+        // We will use whichever age is oldest (largest). This handles the fact that the reference now can get "stuck" if there is no new data in the partition
+        const bufferAge = Math.max(bufferAgeInMemory, bufferAgeFromReference)
+
         logContext['bufferAge'] = bufferAge
```

Comment:
As a nit I'd push more of these times into the log context since we've had such problems with whether something flushes but also the log context is massive so maybe we wait until we see this is a still a problem 🤷

Reply:
Thats a good call actually as it was the logcontext that made us realise this problem in the first place 👍